File: lingting-polaris-grpc/src/main/java/live/lingting/polaris/grpc/resolver/PolarisNameResolverFactory.java

package live.lingting.polaris.grpc.resolver;

import com.tencent.polaris.client.api.SDKContext;
import io.grpc.NameResolverRegistry;

/**
 * @author <a href="mailto:liaochuntao@live.com">liaochuntao</a>
 */
public final class PolarisNameResolverFactory {

	private PolarisNameResolverFactory() {
	}

	public static void init(final SDKContext context) {
		NameResolverRegistry.getDefaultRegistry().register(new PolarisNameResolverProvider(context));
	}

}

File: lingting-polaris-grpc/src/main/kotlin/live/lingting/polaris/grpc/resolver/PolarisNameResolverFactory.kt
package live.lingting.polaris.grpc.resolver

import com.tencent.polaris.client.api.SDKContext
import io.grpc.NameResolverRegistry

/**
 * @author [liaochuntao](mailto:liaochuntao@live.com)
 */
object PolarisNameResolverFactory {
    fun init(context: SDKContext?) {
        NameResolverRegistry.getDefaultRegistry().register(PolarisNameResolverProvider(context!!))
    }
}
